src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -817,7 +821,6 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                     assert(argReg == argNode->GetRegNum());
                     argReg = REG_NEXT(argReg);
                 }
-                assert(regCount == 2);
```

Comment:
i think we should validate the number of registers too for each of x2, x3 and x4.

Reply:
Done